lib/lightning/runs/prom_ex_plugin.ex

```diff
@@ -137,17 +140,29 @@ defmodule Lightning.Runs.PromExPlugin do
           event_name: @available_count_event,
           description: "The number of available runs in the queue",
           measurement: :count
+        ),
+        last_value(
+          [:lightning, :run, :queue, :finalised, :count],
+          event_name: @finalised_count_event,
+          description:
+            "The number of runs finalised during the consideration window",
+          measurement: :count
         )
       ]
     )
   end
 
-  def run_queue_metrics(run_age_seconds) do
-    if pid = Process.whereis(Repo) do
-      check_repo_state(pid)
+  def run_queue_metrics(run_age_seconds, consideration_window_seconds) do
+    if repo_pid = Process.whereis(Repo) do
+      check_repo_state(repo_pid)
 
       trigger_run_claim_duration(run_age_seconds)
       trigger_available_runs_count()
+
+      trigger_finalised_runs_count(
+        DateTime.utc_now(),
```

Comment:
since trigger is private and now is used for the threshold it can be inferred that count_finalised_runs will work based on current time. I would stay with two options:
```
A. count_finalised_runs receives the threshold_datetime
B. count_finalised_runs receives the window_seconds

Reply:
@jyeshe Thanks for the suggestion. I pass in `now` for the same reason that I made the `count_finalised_runs` method public rather private: To allow for tests that allow for higher-fidelity and better readability while removing the risk of a test that may flicker in the future. 

By exposing this interface, I can test the primary logic by passing in the same fixed datetime that was used to generate my fixtures and this means that I do not have to add in a more error-prone time tolerance to deal with the fact that the now() used by the production code may have drifted a nontrivial amount in comparison with the now() used to set up my fixtures.

Of course, I am working on the assumption that there is no cheap, non-brittle way to lock the value of now() for an Elixir test (e.g. something similar to Ruby's Timecop). If this assumption is incorrect, I am happy to modify this method to align with your suggestions above.

However, if my assumption is correct, I am hoping that I can convince you that the testing benefits outweigh the cost of the messier method signature?

Reply:
@jyeshe Firstly, my apologies for the above screed :) on closer inspection of the tests I realised that the change you requested will result in far less impact to test fidelity than I expected. In practical terms, the impact is probably zero!

I have made the change as requested in a new commit.